### PR TITLE
Empty cache after phi first attention to support 4k input

### PR DESCRIPTION
**Pull request by hkvision**
Fix https://github.com/analytics-zoo/nano/issues/1348

### CODE REVIEW

**Comment by hkvision** [comment]
Test result is here: https://github.com/analytics-zoo/nano/issues/1348#issuecomment-2102086271
Performance is not impacted. If there are issues, may add this to low memory mode. But at this moment, add this by default for all cases.

File: python/llm/src/ipex_llm/transformers/convert.py [hunk 1]
```diff
@@ -188,6 +188,12 @@ def is_linear_module(module):
     return result, (in_features, out_features, mp_group)
 
 
+def empty_cache_post(module, input, output):
+    try:
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/convert.py:192]: why not check if it's xpu here?

Comment (hkvision) [reply python/llm/src/ipex_llm/transformers/convert.py:192]: will it hurt performance for the xpu check? I will test it.

Comment (hkvision) [reply python/llm/src/ipex_llm/transformers/convert.py:192]: Seems the impact is not obvious, https://github.com/analytics-zoo/nano/issues/1348#issuecomment-2102086271
using xpu check may be slightly slower than using this way, >5ms for 4k first token latency.

Do I need to change to xpu check?

**Review by MeouSker77 — APPROVED** [review]
LGTM after fixing style check

**Review by lalalapotter — APPROVED** [review]
LGTM.

**Comment by hkvision** [comment]
Merge this first. Can further modify if necessary later.

**Comment by hkvision** [comment]
http://10.239.44.136:8888/nightly_perf_gpu/transformer_int4_gpu-results-2024-05-09.html Yesterday's performance drop should not relate to this PR, the same behavior already happens on 0505, need further observation.
If there are issues, can restrict the empty cache to first token only, but from my test result, the impact for rest is ignorable.

**Pull request merged**